src/Integration/Binding/BindingController.cs

```diff
@@ -147,18 +147,21 @@ private void OnBindingStarted()
         private void OnBindingFinished(ProjectInformation projectInformation, bool isFinishedSuccessfully)
         {
             this.IsBindingInProgress = false;
-            this.host.VisualStateManager.ClearBoundProject();
 
             if (isFinishedSuccessfully)
             {
-                this.host.VisualStateManager.SetBoundProject(projectInformation);
+                if (projectInformation.Key != this.host.VisualStateManager.BoundProjectKey)
```

Comment:
It was added because we needed to filter these unnecessary calls when an "Update" was called, but now the check in `HandleSolutionBindingChanges` is using `SolutionBindingInformationProvider`, which doesn't have this problem.